Test/Unit/Helper/ConfigTest.php

```diff
@@ -534,15 +514,266 @@ public function getIgnoredShippingAddressCoupons()
+        }
+    }
+
+    public function isIPRestrictedProvider()
+    {
+        return [
+            ['123.123.123.123', ['123.123.123.123'], false],
+            ['111.111.111.111', ['222.222.222.222', '123.123.123.123'], true],
+        ];
+    }
+
+    /**
+     * @test
+     */
+    public function getAmastyGiftCardConfig()
+    {
+        $aditionalConfig = '{"amastyGiftCard": {"payForEverything": true}}';
+        $this->scopeConfig
+            ->expects(self::once())
+            ->method('getValue')
+            ->with(BoltConfig::XML_PATH_ADDITIONAL_CONFIG, \Magento\Store\Model\ScopeInterface::SCOPE_STORE, null)
+            ->willReturn($aditionalConfig);
+        $amastyGiftCardConfig = $this->currentMock->getAmastyGiftCardConfig();
+        $this->assertTrue($amastyGiftCardConfig->payForEverything);
+    }
+
+    /**
+     * @test
+     */
+    public function shouldAdjustTaxMismatch()
+    {
+        $aditionalConfig = '{"adjustTaxMismatch": false}';
+        $this->scopeConfig
+            ->expects(self::once())
+            ->method('getValue')
+            ->with(BoltConfig::XML_PATH_ADDITIONAL_CONFIG, \Magento\Store\Model\ScopeInterface::SCOPE_STORE, null)
+            ->willReturn($aditionalConfig);
+        $adjustTaxMismatch = $this->currentMock->shouldAdjustTaxMismatch();
+        $this->assertFalse($adjustTaxMismatch);
+    }
+
+    /**
+     * @test
+     */
+    public function getToggleCheckout()
+    {
+        $aditionalConfig = '
+        {
+            "toggleCheckout": {
+                "active": true,
+                "magentoButtons": [
+                    "#top-cart-btn-checkout",
+                    "button[data-role=proceed-to-checkout]"
+                ],
+                "showElementsOnLoad": [
+                    ".checkout-methods-items",
+                    ".block-minicart .block-content > .actions > .primary"
+                ],
+                "productRestrictionMethods": [
+                    "getSubscriptionActive"
+                ],
+                "itemRestrictionMethods": [
+                    "getIsSubscription"
+                ]
+            }
+        }';
+        $this->scopeConfig
+            ->expects(self::once())
+            ->method('getValue')
+            ->with(BoltConfig::XML_PATH_ADDITIONAL_CONFIG, \Magento\Store\Model\ScopeInterface::SCOPE_STORE, null)
+            ->willReturn($aditionalConfig);
+        $toggleCheckout = $this->currentMock->getToggleCheckout();
+        $this->assertTrue($toggleCheckout->active);
+        $this->assertArraySimilar(
+            ["#top-cart-btn-checkout", "button[data-role=proceed-to-checkout]"],
+            $toggleCheckout->magentoButtons
+        );
+    }
+
+    private function getPageFilters($aditionalConfig = null)
+    {
+        $aditionalConfig = $aditionalConfig ?: '
+        {
+            "pageFilters": {
+                "whitelist": ["checkout_cart_index", "checkout_index_index", "checkout_onepage_success"],
```

Comment:
have a global variable to tie this to the function below as well

Reply:
@ethanwayda22 all done